### ISSUE TITLE
Client Freeze on startup

### ISSUE THREAD

**Issue by schnotzler** [mezz/JustEnoughItems#758]
I cant seem to find the cause of this crash: One of my mods seems to add a broken Recipe, but since the recipe uses only vanilla items and is just mentioned as _net.minecraft.item.crafting.ShapedRecipes@3680b89b._
I cant find the mod. It would be great if someone could tell me which mod is causing the problem.
It would be very time consuming for me to go by trial and error, since my pack includes 250+ mods :/
[Minecraft Log](https://paste.ee/p/E7jj0)   There is no crashreport, because minecraft just freezes

**Comment by mezz** [comment]
Thanks for the report!
I am working to address this with a PR to Forge. Please see this issue for more info: https://github.com/mezz/JustEnoughItems/issues/749

**Comment by mezz** [comment]
Note that these broken recipes are not a crash, just a log. It may just cause a small delay during loading but your game should still run.

**Comment by schnotzler** [comment]
Thank you for your Reply. After ~25 mins it continues loading, you were correct.

**Comment by mezz** [comment]
Hm that's way too long.
You should check the mod list in the previous issue and see what mods you have in common. Hopefully it will narrow things down a bit.
Alternatively I can make a custom Forge build you can try, but using a custom Forge with your MultiMC pack is [slightly complicated](https://github.com/MultiMC/MultiMC5/issues/1782).

**Comment by enricotagliavini** [comment]
I've got this problem as well see my [minecraft log](https://paste.ee/p/osiQP).

The interesting part where it seems to get stuck is

```
[13:45:59] [Client thread/INFO] [JEI]: Building item filter...
[13:48:57] [Client thread/INFO] [JEI]: Built	item filter in 177879 ms
```

Normally this takes 4 seconds
```
[13:59:38] [Client thread/INFO] [JEI]: Building item filter...
[13:59:42] [Client thread/INFO] [JEI]: Built	item filter in 3838 ms
```

That's a long time and it doesn't even look like it's using much CPU during this phase (barely 10% of one core).

I'll add the following as it might be intersting. All mods and forge (version 12.18.3.2221) are kept at the same version except jei and Immersive Engineering.

Working (well without a massive slowdown at game load) configuration:
- Immersive Engineering 0.10-56
- JEI 3.14.5.406

Configuration with significant delay at game startup:
 - Immersive Engineering 0.10-57 (requires a newer version of JEI, thus I updated JEI as well)
 - JEI 3.14.7.415

So I tried with newer JEI and older IE, still has the slowdown during game load
Configuration with significant delay at game startup:
 - Immersive Engineering 0.10-56
 - JEI 3.14.7.415

So basically just changing JEI leaving all other mods at the same version can show the issue. If there are broken recipes they are present also with the older version, but the delay is not present.

If it help I can try to find the latest version of JEI that works and thus the first one presenting the issue.

Or do you recommend waiting for newer forge?

**Comment by schnotzler** [comment]
I dont know where i found this solution anymore, but for me it helped to change the enabled option to false in config\splash.properties. I forgot what mod this was from and where i found it. Re-enabling it slows the -Building item filter- down again.

Maybe it helps you too.

The broken recipes dont seem to cause the slowdown at all.

**Comment by enricotagliavini** [comment]
As this sounds all very odd I checked and release 412 is the first one presenting the problem, which should mean f180dfcbaa24abdbdf0afd78546ef9b310f4d2b1 is the one causing the issue. Hopefully it's as simple as some leftover code not removed with the commit and can be solved quickly. If not, can this be reverted until a fix is found please? Also should the issue be reopened since it doesn't look like a duplicate any more?

I can confirm disabling the splash is a workaround for the issue. However I consider it cumbersome for normal gamers. I play with a few friends and it's hard enough for them to install modded minecraft, editing a config file in a very odd (for normal users standard) location is very hard. Thank you anyway for mentioning this is useful information.

Btw config\splash.properties is from Forge itself I suppose, not from a mod.

**Comment by luffykick** [comment]
try this fix found here ...https://forum.feed-the-beast.com/threads/stuck-on-loading-6-7-jei-3-0-7.218851/

**Comment by enricotagliavini** [comment]
@luffykick there is no fix mentioned there. There is the workaround, already mentioned, about disabling the spash screen. I'll eventually surrender and disable it, but I would prefer not to indeed.

However there is a useful link to the bug report. Vsync issue... well that's bad. Vsync is required to have tear free experience, that's why I can see the issue (I have it disabled in Minecraft but I have it enabled as a whole for the Desktop). So a Forge bug in the end

**Comment by mezz** [comment]
You guys are talking about a different bug now, vsync is this one: https://github.com/mezz/JustEnoughItems/issues/746

**Comment by enricotagliavini** [comment]
Yeah I realized now. Sorry for the noise

**Comment by bookerthegeek** [comment]
@schnotzler, https://github.com/MinecraftModDevelopmentMods/BetterAgriculture/issues/43
